File: ios/MassiveRPG/Domain/UseCase/GetCharacterListUseCaseImpl.swift
//
//  GetCharacterListUseCaseImpl.swift
//  MassiveRPG-Prod
//
//  Created by Onur Var on 10.04.2024.
//

class GetCharacterListUseCaseImpl: GetCharacterListUseCase {
    // MARK: Variables

    let characterRepository: CharacterRepository

    // MARK: Life Cycle

    init(characterRepository: CharacterRepository) {
        self.characterRepository = characterRepository
    }

    // MARK: Methods

    func execute() async -> Result<[Character], Error> {
        return await characterRepository.getList()
    }
}

File: android/app/src/main/java/com/onurvar/massiverpg/domain/usecase/GetCharacterListUseCaseImpl.kt

package com.onurvar.massiverpg.domain.usecase

import com.onurvar.massiverpg.domain.model.Character
import com.onurvar.massiverpg.domain.protocol.repository.CharacterRepository
import com.onurvar.massiverpg.domain.protocol.usecase.GetCharacterListUseCase

class GetCharacterListUseCaseImpl(
    private val characterRepository: CharacterRepository
) : GetCharacterListUseCase {
    override suspend fun execute(): List<Character> {
        return characterRepository.getCharacterList()
    }
}